### DISCUSSION TITLE
How to enable WalletConnect to directly connect to Binance Smart Chain instead of Ethereum?

### DISCUSSION THREAD

**Discussion by devturi** [WalletConnect/walletconnect-monorepo#1207] [Developer Support]
I am working on a DAPP which works on Binance Smart Chain. Now, when I scan the qrcode with trust wallet, it shows Ethereum Chain by default and I have to change it manually to Binance Smart Chain from the option. However, this does not look good. Normally on metamask, I can use  await window.ethereum.request({
      method: "wallet_addEthereumChain",
      params: [
        {
          ...networks2[networkName],
        },
      ],
    });
  } catch (err) {
    console.error(err);
    // setError(err.message);
  }

However, on trust wallet, I dont get to change the network once walletConnect connection is established. So, I need to enable BSC before the connection. Can anyone please help?

**Comment by tomisinteazer** [comment]
is there an update on this I'm facing the exact same issue

**Reply by tomisinteazer** [reply]
pancake swap was able to achieve this so definitely it can be done

**Reply by Lukas439** [reply]
I think this is able if you try Binance wallet

**Comment by devturi** [comment]
Actually, I also found the solution for it. It was kind of a small trick.
Lol. I actually bruteforced all of the solutions that I had in mind and
finally one thing worked

**Reply by tomisinteazer** [reply]
i found a solution i posted my findings in  #1208 
however i would still like to know how you solved your side of the bargain
